internal/painter/software/draw_test.go

```diff
@@ -35,6 +35,26 @@ func TestDrawImage(t *testing.T) {
 	test.AssertImageMatches(t, "draw_image_default.png", target)
 }
 
+func TestDrawImageNearestFilter(t *testing.T) {
```

Comment:
These tests are what we needed I think, thanks. 
The output images we compare with make the value of this clear :)